### PR TITLE
Add button for showing/hiding graph in connections

### PR DESCRIPTION
**Pull request by borisvida**

### CODE REVIEW

File: client/src/components/Connections/components/Results.js [hunk 1]
```diff
@@ -1,25 +1,43 @@
...
+    </Collapse>
     {props.connections.map((connEid) => <InfoLoader key={connEid} eid={connEid} hasConnectLine />)}
   </div>
 )
 
 export default compose(
   EntitySearchWrapper,
+  withState('graphShown', 'setGraphShown', false),
+  withHandlers({
+    toggleGraphShown: ({setGraphShown, graphShown}) => () => setGraphShown(!graphShown),
```
Comment (jakubukrop) [inline client/src/components/Connections/components/Results.js]: Add type `StateProps` to function args.

**Comment by jakubukrop** [comment]
When the graph is opened its layout is broken (out of viewport). Try http://localhost:3000/prepojenia?eid1=Miro%20Noga&eid2=%C5%A0tefan%20Skr%C3%BAcan%C3%BD
Graph should not be in DOM when it's not shown (moreover, it may potentially preload lots of data).
This quickfix `{props.graphShown && <Subgraph preloadNodes {...props} />}`
solves the preloading problem and initial graph display - but when you click hide & show again the graph layout is broken again. I don't know how to fix that.
(Though we may still merge this PR with the above quickfix and create issue for the "repeated hide-show problem" ...)

File: client/src/components/Connections/components/Results.js (hunk 1 above)
Comment (borisvida) [reply client/src/components/Connections/components/Results.js]: I am not sure what you mean. I am defining `toggleGraphShown` prop here, `setGraphShown` is another method, that isn't included in `StateProps`.

**Comment by borisvida** [comment]
So, I removed the `Collapse` element and it solved the broken layout. The opening of the graph is not so nice and smooth anymore, but I really think it's better than having to drag the graph back into viewport. Thanks for the catch!

File: client/src/components/Connections/components/Results.js (hunk 1 above)
Comment (jakubukrop) [reply client/src/components/Connections/components/Results.js]: Oh, you are right ... still it would be nice to add types for those args.

**Pull request merged**